### PR TITLE
Saraswathi 1242

### PR DESCRIPTION
**Pull request by Saraswathi-classic**
Home Page
-------------
image

Cart Page
-----------
image

Empty Cart
-------------
image

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,2592 @@
+{
```
Comment (atmeshwar-classic) [inline package-lock.json:1]: We already have yarn.lock file and having 2 lockfiles is considered a bad practice.
Please remove it and use yarn to install dependencies.

File: src/App.tsx [hunk 2]
```diff
@@ -1,34 +1,21 @@
...
-        Click on the Vite and React logos to learn more
-      </p>
-    </div>
+    <Router>
+      <Navbar />
+      <Routes>
+        <Route path="/" element={<BooksPage />} />
+        <Route path="/cart" element={<CartPage />} />
+      </Routes>
+    </Router>
```
Comment (atmeshwar-classic) [inline src/App.tsx]: Your routes should be inside its own file and imported here. For eg:
src/routes/routes.ts

We should always follow the proper structure.

File: src/components/BookCard/BookCard.tsx [hunk 3]
```diff
@@ -7,7 +11,22 @@ import type { Book } from '../../domain/books/types'
...
\ No newline at end of file
+    <div className="book-card">
+      <div className="book-card-img">
+        <img alt="Book Image" />
+      </div>
+      <div className="book-card-contents">
+        <p style={{ fontSize: "1.1em" }}>{props.name}</p>
+        <p style={{ fontSize: "25px" }}>{props.author}</p>
+        <p style={{ fontSize: "22px" }}>{props.description}</p>
+        <p style={{ fontSize: "20px" }}>${props.price}</p>
```
Comment (atmeshwar-classic) [inline src/components/BookCard/BookCard.tsx]: These styles should have been defined in BookCard.css

File: src/components/BookCard/BookCard.tsx [hunk 4]
```diff
@@ -7,7 +11,22 @@ import type { Book } from '../../domain/books/types'
 type BookProps = Book;
 
 export const BookCard = (props: BookProps): JSX.Element => {
+  const dispatch: AppDispatch = useDispatch();
+
```
Comment (atmeshwar-classic) [inline src/components/BookCard/BookCard.tsx]: Mostly if not always try to destructure component props.
`({name, author, description, price}: BookProps)`

File: src/domain/cart/CartPage.tsx [hunk 5]
```diff
@@ -1,5 +1,50 @@
...
+      <>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Author</th>
+            <th>Description</th>
+            <th>Price</th>
+            <th>Quantity</th>
+            <th>Action</th>
+          </tr>
```
Comment (atmeshwar-classic) [inline src/domain/cart/CartPage.tsx:27]: Using a table was good touch 😄

File: src/store/store.ts [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
+  books: booksSliceReducer,
+  cart: cartSliceReducer
 });
 
 export const store = configureStore({
   reducer: combinedReducer
-})
+});
+
+store.dispatch(getBooks());
```
Comment (atmeshwar-classic) [inline src/store/store.ts]: Do not dispatch this action here. We should call it inside BooksPage's useEffect.

File: src/components/BookCard/BookCard.tsx (hunk 3 above)
Comment (Saraswathi-classic) [reply src/components/BookCard/BookCard.tsx]: Fixed

File: src/components/BookCard/BookCard.tsx (hunk 4 above)
Comment (Saraswathi-classic) [reply src/components/BookCard/BookCard.tsx]: Fixed

File: src/store/store.ts (hunk 6 above)
Comment (Saraswathi-classic) [reply src/store/store.ts]: Fixed

File: package-lock.json (hunk 1 above)
Comment (Saraswathi-classic) [reply package-lock.json:1]: Fixed

File: src/App.tsx [hunk 7]
```diff
@@ -1,21 +1,19 @@
...
-
   return (
     <Router>
       <Navbar />
       <Routes>
-        <Route path="/" element={<BooksPage />} />
-        <Route path="/cart" element={<CartPage />} />
+        {routes.map(({ path, Component }) => (
+          <Route path={path} element={<Component />} />
+        ))}
```
Comment (atmeshwar-classic) [inline src/App.tsx]: I think you have not understood what I was saying.
In, src/router/router.tsx file, you should be placing the routes.

`export const router = () => {
  <Routes>
   <Route path="/" component={<BooksPage />} />
   </Routes>
|
`

File: src/routes/routes.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+export const routes = [
+    {
+        path: "/",
+        Component: BooksPage
+    },
+    {
+        path: "/cart",
+        Component: CartPage
+    }   
+]
```
Comment (atmeshwar-classic) [inline src/routes/routes.ts]: Please see the above comment

File: src/App.tsx (hunk 7 above)
Comment (Saraswathi-classic) [reply src/App.tsx]: Okay

Comment (Saraswathi-classic) [reply src/App.tsx]: Fixed

**Comment by atmeshwar-classic** [comment]
I am closing this PR as the assessment is complete.
Please keep the pointers I shared in mind @Saraswathi-classic .

**Comment by Saraswathi-classic** [comment]
Sure, will keep all the points in mind which you have suggested.
Thanks @atmeshwar-classic

**Pull request closed without merging**